### PR TITLE
Rhorsey/sampling v2 plotting patch

### LINKED ISSUE #216: Postprocessing errors
#186 introduced an post-processing errors; unclear on how many. This is a blocker for post-processing EUSS runs.
First error encountered in LazyFrame:
```
Traceback (most recent call last):
  File "C:\Users\mdahlhau\Documents\GitHub\NREL\comstock\postprocessing\compare_upgrades.py", line 63, in <module>
    main()
  File "C:\Users\mdahlhau\Documents\GitHub\NREL\comstock\postprocessing\compare_upgrades.py", line 54, in main
    cbecs.export_to_csv_wide()  # May comment this out after run once
  File "C:\Users\mdahlhau\Documents\GitHub\NREL\comstock\postprocessing\comstockpostproc\cbecs.py", line 486, in export_to_csv_wide
    self.data.to_csv(file_path, index=False)
AttributeError: 'LazyFrame' object has no attribute 'to_csv'
```

### PR DESCRIPTION
**Pull request by rHorsey**
Pull request overview
---------------------

Updates plotting scripts for comparing upgrades, baselines, and CBECS. AMI and Timeseries are up next - EIA requires a transition to our current postproc paradigm.

 - Fixes #216 although I'm pretty sure the baseline there isn't correct, thus I've added a 10k sample to sampling/resources.

### Pull Request Author

This pull request makes changes to (select all the apply):
 - [ ] Documentation
 - [ ] Infrastructure (includes apptainer image, buildstock batch, dependencies, continuous integration tests)
 - [X] Sampling
 - [ ] Workflow Measures
 - [ ] Upgrade Measures
 - [ ] Reporting Measures
 - [X] Postprocessing

Author pull request checklist:

 - [X] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [ ] ~Added tests for new measures~
 - [ ] ~Updated measure .xml(s)~
 - [ ] ~Register values added to `comstock_column_definitions.csv`~
 - [ ] ~Both `options_lookup.tsv` files updated~
 - [X] 10k+ test run
 - [ ] ~Change documentation written~
 - [ ] ~Measure documentation written~
 - [ ] ~ComStock documentation updated~
 - [ ] ~Changes reflected in example `.yml` files~
 - [ ] ~Changes reflected in `README.md` files~
 - [X] Added 'See ComStock License' language to first two lines of each code file
 - [ ] ~Implements corresponding measure tests and indexing path in `test/measure_tests.txt` or/and `test/resource_measure_tests.txt`~
 - [ ] ~All new and existing tests pass the CI~

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a code review on GitHub
 - [ ] All related changes have been implemented: data and method additions, changes, tests
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] Reviewed change documentation
 - [ ] Ensured code files contain License reference
 - [ ] Results differences are reasonable
 - [ ] Make sure the newly added measures has been added with tests and indexed properly
 - [ ] CI status: all tests pass

#### ComStock Licensing Language - Add to Beginning of Each Code File
```
# ComStock™, Copyright (c) 2023 Alliance for Sustainable Energy, LLC. All rights reserved.
# See top level LICENSE.txt file for license terms.
```

### CODE REVIEW

**Comment by wnykuang** [comment]
Looking

File: postprocessing/comstockpostproc/comstock.py [hunk 1]
```diff
@@ -1992,21 +2050,29 @@ def create_geospatially_resolved_aggregations(self, geographic_col_name, pretty_
...
 
         # Export dictionaries corresponding to the exported columns
         self.export_data_and_enumeration_dictionary()
 
+        # Return the geospatially aggregated dataframe for use by plotting, etc.
+        return spatial_aggregation
 
     def add_weights_aportioned_by_stock_estimate(self, apportionment: Apportion, keep_n_per_apportionment_group=False):
+        # This function doesn't support already CBECS-weighted self.data - error out
+        if self.CBECS_WEIGHTS_APPLIED:
```
Comment (wenyikuang) [inline postprocessing/comstockpostproc/comstock.py:2087]: GREAT!

Comment (wenyikuang) [reply postprocessing/comstockpostproc/comstock.py:2065]: Actually there is a even shorter way:

assert self.CBECS_WEIGHTS_APPLIED == True , "Unable to apply apportionment weighting after CBECS weighting - reverse order."

File: postprocessing/comstockpostproc/comstock.py [hunk 2]
```diff
@@ -2478,7 +2548,8 @@ def rmv_units(c):
             assert new.startswith(old)
             logger.debug(f'{old} -> {new}')
 
-        self.data = self.data.rename(crnms)
+        lazyframe = lazyframe.rename(crnms)
+        return lazyframe
```
Comment (wenyikuang) [inline postprocessing/comstockpostproc/comstock.py:2574]: `return lazyframe.rename(crnms)` will be exctaly same.

**Review by wenyikuang — DISMISSED** [review]
Looks good, but I tried to run comstock_to_cbecs_comparison.py with the default template but not works due to `eulp/euss_com/cycle_4_sampling_test_rand_985932_20240321/cycle_4_sampling_test_rand_985932_20240321/` is not existed, the right path should be `cycle_4_sampling_test_rand_985932_20240321/rand_985932_20240321/baseline/`

File: postprocessing/compare_upgrades.py.template [hunk 3]
```diff
@@ -1,63 +1,73 @@
...
+    # Scale ComStock run to CBECS 2018 AND remove non-ComStock buildings from CBECS
+    comstock.add_national_scaling_weights(cbecs, remove_non_comstock_bldg_types_from_cbecs=True)
+
+    # Export CBECS and ComStock data to wide and long formats for Tableau and to skip processing later
+    # cbecs.export_to_csv_wide()  # May comment this out after run once
+    # comstock.create_national_aggregation()
+    # comstock.create_geospatially_resolved_aggregations(comstock.STATE_ID, pretty_geo_col_name='state_id')
+    # comstock.create_geospatially_resolved_aggregations(comstock.COUNTY_ID, pretty_geo_col_name='county_id')
+    # TODO Long is def not working as expected anymore...
+    # comstock.export_to_csv_long()  # Long format useful for stacking end uses and fuels
```
Comment (ChristopherCaradonna) [inline postprocessing/compare_upgrades.py.template:66]: Where is ComStock wide export?

**Comment by rHorsey** [comment]
Thanks @wenyikuang - updated the paths and also switched to using Resource instead of Client in boto3 to avoid an error with the policy on stuff uploaded from Kestrel using the default upload policy.

**Comment by rHorsey** [comment]
@ChristopherCaradonna  can you delete the lighting upgrade in you comstock_data and try rerunning compare_upgrades again? If that works as expected can you review / approve please?

**Review by ChristopherCaradonna — CHANGES_REQUESTED** [review]
@rHorsey Postprocessing completed, most plots look good.

2 Issues:
1) Census Division plots show no energy / square footage in "Middle Atlantic".
2) Non-breaking error message, as show below
Bonus) I get an error for not having 'import json' in the s3_utilties_mixin.py

--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\ccaradon\AppData\Local\anaconda3\envs\comstockpostproc\lib\logging\__init__.py", line 1083, in emit
    msg = self.format(record)
  File "C:\Users\ccaradon\AppData\Local\anaconda3\envs\comstockpostproc\lib\logging\__init__.py", line 927, in format
    return fmt.format(record)
  File "C:\Users\ccaradon\AppData\Local\anaconda3\envs\comstockpostproc\lib\logging\__init__.py", line 663, in format
    record.message = record.getMessage()
  File "C:\Users\ccaradon\AppData\Local\anaconda3\envs\comstockpostproc\lib\logging\__init__.py", line 367, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "C:\Users\ccaradon\Documents\GitHub\ComStock\postprocessing\compare_upgrades.py", line 73, in <module>
    main()
  File "C:\Users\ccaradon\Documents\GitHub\ComStock\postprocessing\compare_upgrades.py", line 58, in main
    comstock.add_national_scaling_weights(cbecs, remove_non_comstock_bldg_types_from_cbecs=True)
  File "C:\Users\ccaradon\Documents\GitHub\ComStock\postprocessing\comstockpostproc\comstock.py", line 1900, in add_national_scaling_weights
    logger.info("sf wt_area_col shape: ", sf[wt_area_col].shape)
Message: 'sf wt_area_col shape: '
Arguments: ((14,),)

File: postprocessing/compare_upgrades.py.template (hunk 3 above)
Comment (rHorsey) [reply postprocessing/compare_upgrades.py.template:66]: That's the create_national_aggregations method now - same code under the hood but redone to get correctly weighted national level results!

**Comment by rHorsey** [comment]
@ChristopherCaradonna 's issues addressed - ready for approval! Great catch on mid-atlantic - many thanks!.

**Review by ChristopherCaradonna — APPROVED** [review]
Changes look good, approved.

**Pull request merged**